### PR TITLE
Add support for custom allocators by the allocator-api2 crate.

### PR DESCRIPTION
**Pull request by tage64**
Due to the disclaimer about new contributions in the README, you might think this PR is too much.
However, I needed it for my use case and thought that I can at least share it and then it is up to
you (the maintainer) to decide what to do with it.

This commit adds allocator-api2 (a drop-in replacement for the
`Allocator` trait on stable) as a dependency. The underlying `Vec` is
replaced by `allocator_api2::vec::Vec`, and the type signiture of
`VecMap<T>` is changed to `VecMap<T, A: Allocator = Global>`.

It adds the methods `new_in()` and `with_capacity_in()` which creates a
`VecMap` with a custom allocator. The old `new()` and `with_capacity()`
methods are not removed and defaults to the default global allocator.

I think that this commit is backwards compatable.

### CODE REVIEW

**Comment by pczarn** [comment]
Hey. That's ok. I will try to pick up maintenance sometime like I did for `bit-vec`

**Comment by pczarn** [comment]
Huh, have I pushed into your `master`?

**Comment by pczarn** [comment]
Looks good. Thanks

File: src/lib.rs [hunk 1]
```diff
@@ -734,29 +768,29 @@ impl<V: Clone> Clone for VecMap<V> {
     }
 }
 
-impl<V: PartialEq> PartialEq for VecMap<V> {
```
Comment (pczarn) [inline src/lib.rs:737]: What's the story for comparison with regard to allocators?

Comment (tage64) [reply src/lib.rs:737]: After looking at the `Vec` in the standard library, it seems as that `PartialEq` and `PartialOrd` can be implemented for different allocators. I tried and it works so I will make a new pr with these changes.

**Pull request merged**